Firestore/core/test/unit/remote/bloom_filter_test.cc

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/remote/bloom_filter.h"
+#include <vector>
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace remote {
+
+using util::Status;
+using util::StatusOr;
+
+class BloomFilterTest : public ::testing::Test {};
+
+TEST_F(BloomFilterTest, CanInstantiateEmptyBloomFilter) {
+  BloomFilter bloom_Filter = BloomFilter(std::vector<uint8_t>{}, 0, 0);
+  EXPECT_EQ(bloom_Filter.bit_count(), 0);
+}
+
+TEST_F(BloomFilterTest, CanInstantiateNonEmptyBloomFilter) {
+  {
+    BloomFilter bloom_Filter_1 = BloomFilter(std::vector<uint8_t>{1}, 0, 1);
+    EXPECT_EQ(bloom_Filter_1.bit_count(), 8);
+  }
+  {
+    BloomFilter bloom_Filter_2 = BloomFilter(std::vector<uint8_t>{1}, 7, 1);
+    EXPECT_EQ(bloom_Filter_2.bit_count(), 1);
+  }
+}
+
+/* Test Create factory method with valid and invalid inputs */
+TEST_F(BloomFilterTest, CreateMethodShouldReturnBloomFilterOnValidInputs) {
+  StatusOr<BloomFilter> maybe_bloom_filter =
+      BloomFilter::Create(std::vector<uint8_t>{1}, 1, 1);
+  EXPECT_TRUE(maybe_bloom_filter.ok());
```

Comment:
Use `ASSERT_TRUE` rather than `EXPECT_TRUE` when checking the return value of `Status::ok()`. This is because if it _were_ to return the wrong value then the test should end immediately, because calling `ValueOrDie()` is going to ... die ... and crash the tests, which is undesirable.

Please make this change here and elsewhere in this file. 

The rule of thumb is: if the test can meaningfully continue if the assertion fails, then use `EXPECT_XXX`; otherwise, use `ASSERT_XXX`.